src/EventStore.Core.Tests/Services/Transport/Tcp/TcpClientDispatcherTests.cs

```diff
@@ -295,5 +281,212 @@ private EventRecord CreateLinkEventRecord() {
 				"link-stream", -1, PrepareFlags.SingleWrite | PrepareFlags.Data, SystemEventTypes.LinkTo,
 				Encoding.UTF8.GetBytes(string.Format("{0}@test-stream", long.MaxValue)), new byte[0]), "link-stream", SystemEventTypes.LinkTo);
 		}
+
+		[Test]
+		public void unwraps_ReadEvent() {
+			var dto = new ReadEvent(
+				eventStreamId: "my-stream",
+				eventNumber: 5,
+				resolveLinkTos: true,
+				requireLeader: true);
+
+			var package = new TcpPackage(TcpCommand.ReadEvent, Guid.NewGuid(), dto.Serialize());
+
+			var message = _dispatcher.UnwrapPackage(
+				package: package,
+				envelope: new NoopEnvelope(),
+				user: SystemAccounts.System,
+				tokens: new Dictionary<string, string>(),
+				connection: null,
+				version: (byte)ClientVersion.V2);
+
+			Assert.IsInstanceOf<ClientMessage.ReadEvent>(message);
+			var readEvent = message as ClientMessage.ReadEvent;
+			Assert.AreEqual("my-stream", readEvent.EventStreamId);
+			Assert.AreEqual(5, readEvent.EventNumber);
+			Assert.AreEqual(true, readEvent.ResolveLinkTos);
+			Assert.AreEqual(true, readEvent.RequireLeader);
+			Assert.That(readEvent.Expires, Is.EqualTo(DateTime.UtcNow + _readTimeout).Within(_tolerance));
+		}
+
+		[Test]
+		public void unwraps_ReadStreamEventsForward() {
+			var dto = new ReadStreamEvents(
+				eventStreamId: "my-stream",
+				fromEventNumber: 5,
+				maxCount: 6,
+				resolveLinkTos: true,
+				requireLeader: true);
+
+			var package = new TcpPackage(TcpCommand.ReadStreamEventsForward, Guid.NewGuid(), dto.Serialize());
+
+			var message = _dispatcher.UnwrapPackage(
+				package: package,
+				envelope: new NoopEnvelope(),
+				user: SystemAccounts.System,
+				tokens: new Dictionary<string, string>(),
+				connection: null,
+				version: (byte)ClientVersion.V2);
+
+			Assert.IsInstanceOf<ClientMessage.ReadStreamEventsForward>(message);
+			var readEvent = message as ClientMessage.ReadStreamEventsForward;
+			Assert.AreEqual("my-stream", readEvent.EventStreamId);
+			Assert.AreEqual(5, readEvent.FromEventNumber);
+			Assert.AreEqual(6, readEvent.MaxCount);
+			Assert.AreEqual(true, readEvent.ResolveLinkTos);
+			Assert.AreEqual(true, readEvent.RequireLeader);
+			Assert.That(readEvent.Expires, Is.EqualTo(DateTime.UtcNow + _readTimeout).Within(_tolerance));
+		}
+
+		[Test]
+		public void unwraps_ReadStreamEventsBackward() {
+			var dto = new ReadStreamEvents(
+				eventStreamId: "my-stream",
+				fromEventNumber: 5,
+				maxCount: 6,
+				resolveLinkTos: true,
+				requireLeader: true);
+
+			var package = new TcpPackage(TcpCommand.ReadStreamEventsForward, Guid.NewGuid(), dto.Serialize());
```

Comment:
should TcpCommand.ReadStreamEventsForward be TcpCommand.ReadStreamEventsBackward?

Reply:
yes thanks, i'll fix